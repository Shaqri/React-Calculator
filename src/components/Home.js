import React from 'react';
import Navbar from './Navbar';
import HomeContainer from './HomeContainer';

const Home = () => (
  <div className="home">
    <Navbar />
    <HomeContainer />
  </div>
);

export default Home;
