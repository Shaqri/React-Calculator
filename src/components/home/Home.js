import React from 'react';
// import PropTypes from 'prop-types';
import Navbar from './Navbar';
// import HeroText from './HeroText';

const Home = () => (
  <div className="home">
    <Navbar />
  </div>
);

export default Home;
