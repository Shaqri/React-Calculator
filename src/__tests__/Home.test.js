import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import Home from '../components/Home';


jest.mock('../components/Navbar', () => () => <nav data-testid='navbar' />) ;

describe('Home', () => {

  it('should render navbar component', async () => {
    render(<Home />);
    expect(await screen.getByTestId(/navbar/)).toBeInTheDocument();
  });
  
  it('should render a welcome message in home header', async () => {
    render(<Home />);

    expect( await screen.findByText('Welcome to our page')).toMatchSnapshot();
  });
});
