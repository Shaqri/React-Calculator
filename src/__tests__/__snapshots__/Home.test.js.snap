// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a welcome message in home header 1`] = `
<h1>
  Welcome to our page
</h1>
`;
